### PR TITLE
fix(eslint-plugin): [prefer-optional-chain] handle binary expressions in negated or

### LINKED ISSUE #5991: Bug: [prefer-optional-chain] Unexpected member property type: Identifier
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.8.4&sourceType=module&code=CYUwxgNghgTiAEYD2A7AzgF3gSwFzyhQE8BuAWAChRJYFl0tYYoj9DTLKBCJlgbWzwA1PACMAXXgAfKfB4xmRAcPgAmcZSA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WloHsalfkwCG8WmQAWo5uii9o-aJHBgAviHVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3QAacDUjoAHoizJ46IngCGqACL9YmNXl3Cwo9FIC+IM0A

### Repro Code

```typescript
declare const i: any;
declare const array: any;

!array[i + 1] || !array[i + 2]
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/prefer-optional-chain": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true,
    "experimentalDecorators": true
  }
}
```

### Expected Result

No errors

### Actual Result

Unexpected member property type: Identifier

### Additional Info

This is a regression of 5.43.0. In version 5.42.1 all works correctly.

### Versions

| package | version |
| -- | -- |
| `@typescript-eslint/eslint-plugin` | `5.43.0` |
| `@typescript-eslint/parser` | `5.43.0` |
| `TypeScript` | `4.8.4` |
| `ESLint` | `8.15.0` |
| `node` | `web` |

### PR DESCRIPTION
**Pull request by omril1**
## PR Checklist

- [x] Addresses an existing open issue: fixes #5991
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

## Overview

I'm not sure why the default switch case is to throw instead of ignoring the code 😢 
~The issue was not yet triaged but I opened a PR just in case~

### CODE REVIEW

File: packages/eslint-plugin/src/rules/prefer-optional-chain.ts [hunk 1]
```diff
@@ -407,6 +407,10 @@ export default util.createRule({
             propertyText = getMemberExpressionText(node.property);
             break;
 
+          case AST_NODE_TYPES.BinaryExpression:
+            propertyText = sourceCode.getText(node.property);
+            break;
```
Comment (JoshuaKGoldberg) [inline packages/eslint-plugin/src/rules/prefer-optional-chain.ts:412]: There's already a `case` a few lines above for `AST_NODE_TYPES.Literal` and `AST_NODE_TYPES.TemplateLiteral`. Could you add this there please?

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Looks great in general, thanks for sending this in so quickly! Just requesting changes for a little code simplification please. 🎉

**Comment by RobYed** [comment]
@omril1 Thanks for the quick fix!

@JoshuaKGoldberg Do you have an estimate of when this fix will be released? My project is also affected by this issue, thus looking forward to the next release.

**Comment by bradzacher** [comment]
@RobYed https://github.com/typescript-eslint/typescript-eslint/blob/main/README.md#versioning

**Comment by RobYed** [comment]
@bradzacher Thanks! I must have been blind! 🙈

**Pull request merged**